kjourney/src/app/register-user/register-user.component.ts

```diff
@@ -0,0 +1,128 @@
+import { Injectable } from '@angular/core';
+import { RegisterUserModel } from './register-user-model';
+import { Component, OnInit, Input, EventEmitter } from '@angular/core';
+import { FormGroup, FormBuilder, Validators, ControlContainer } from '@angular/forms';
+import { AuthenticationService } from '../_services/authentication.service';
+import { User } from '../user';
+import { HttpClient, HttpHeaders,  HttpResponse } from '@angular/common/http';
+import { Router, ActivatedRoute } from '@angular/router';
+
+import { first } from 'rxjs/operators';
+
+
+
+@Injectable({ providedIn: 'root' })
+@Component({
+  selector: 'app-register-user',
+  templateUrl: './register-user.component.html',
+  styleUrls: ['../login/login.component.css']
+})
+export class RegisterUserComponent implements OnInit {
+  public regUser : RegisterUserModel;
+  public jobDropDownEnabled : boolean = false;
+  public jobRoleDropDownEnabled : boolean = false;
+  public loginErrorMessage = "";
+
+  form: FormGroup = this.fb.group({
+    username: [null],
+    password: [null],
+    confirmPassword: [null],
+    capDropDown: [null],
+    jobFamDropDown: [null],
+    jobRoleDropDown: [null],
+    fullname: [null]   
+  });
+
+  constructor(
+    private fb: FormBuilder,
+    private http: HttpClient,
+    private authenticationService: AuthenticationService,
+    private router: Router) {}
+
+  // defined the array of data
+  capabilities: Object[] = [
+
+];
+
+  jobFamilies: Object[] = [];
```

Comment:
Arent there models for these?

Reply:
we're expecting an array of {viewValue:x, value:y} returned from the express API

Do we need a model for that?